scipy/optimize/slsqp.py

```diff
@@ -249,17 +254,50 @@ def _minimize_slsqp(func, x0, args=(), jac=None, bounds=None,
         iprint = 0
 
     # Transform x0 into an array.
-    x = asfarray(x0).flatten()
+    full_x0 = np.copy(asfarray(x0).flatten())
 
     # SLSQP is sent 'old-style' bounds, 'new-style' bounds are required by
     # ScalarFunction
     if bounds is None or len(bounds) == 0:
-        new_bounds = (-np.inf, np.inf)
+        b = np.array([np.inf] * len(full_x0))
+        full_new_bounds = -b, b
     else:
-        new_bounds = old_bound_to_new(bounds)
+        full_new_bounds = old_bound_to_new(bounds)
 
     # clip the initial guess to bounds, otherwise ScalarFunction doesn't work
-    x = np.clip(x, new_bounds[0], new_bounds[1])
+    full_x0 = np.clip(full_x0, full_new_bounds[0], full_new_bounds[1])
+
+    # Filter out parameters whose lower_bound == upper_bound.
+    # This definitely needs to be done when jacobians are estimated
+    # numerically. However, even if the user supplies a callable jacobian
+    # filtering is advised because some of the user supplied constraints may
+    # not have callable jacobians.
```

Comment:
Could check for that, right?

Reply:
We could. However, it would require a subsequent change to filter everything (including a double pass through the constraints loop) if one of them didn't have a callable jacobian. I reckon this would be quite complicated.

- I think whilst equal bounds occur, they're not the majority of problems (say 1%).
- If equal bounds do occur there probably won't be callable jacobians for all constraints and the original function (say 10%).

Since the overall probability was so low, I didn't think it'd be worth implementing for the complexity involved.